CI/build_smooth_host.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+. ./CI/prepare_idf.sh
+
+if [ -d build_host ]; then
+	rm -rf build_host
+fi
+mkdir -p build_host
+
+set -e
+
+declare -a tests
+count=0
+
+# Get list of tests
+cd test
+for i in $(ls -d */); do
+  tests[count]=${i%%/}
+  # echo ${test[$count]}
+  count=$((count + 1))
+done
+
+cd ../build_host
+
+cp ../CMakeLists.txt ./original_CMakeLists.txt
+
+for current in "${tests[@]}"; do
+
+    echo "#######################################"
+    echo "Compiling project $current"
+    echo "#######################################"
+    echo
+    cp ./original_CMakeLists.txt ../CMakeLists.txt
+    pattern="s/(selected_test_project)\s+\w+/\1 ${current}/g"
+
+    sed -i -r -e "$pattern" ../CMakeLists.txt
+
+    cmake .. -G "Ninja" -DCMAKE_TOOLCHAIN_FILE=../CI/gnu-toolchain.cmake
+    ninja
+
+done
+
+# Clean up
+cp ./original_CMakeLists.txt ../CMakeLists.txt
```

Comment:
This is just a copy of build_smooth_eps.sh, isn't it? Can we use a single file for both? Unless they diverge somehow I don't see a reason to have two copies of the same script.

Reply:
in the ende you are right, but there are still some differences:
- the esp version is not compiling the linux tests
- the build directory and the toolchain is different

we could merge all this into one script file, but then we need an input argument. I am open for both solutions, as you like it.

Reply:
Right. Lets leave it for now and do something better when/if they these files needs to be updated.